misk-embedded/src/main/kotlin/misk/embedded/EmbeddedMisk.kt

```diff
@@ -0,0 +1,82 @@
+package misk.embedded
+
+import com.google.inject.Injector
+import okio.buffer
+import okio.sink
+import okio.source
+import java.io.File
+import java.net.URLClassLoader
+import java.util.jar.JarFile
+
+/**
+ * Loads a Misk Service in the scope of its own class loader such that it can be accessed by in
+ * another JVM project while avoiding classpath collisions on dependencies. The embedded misk
+ * service is accessed by the host project using an "embedding interface". Use this embedding
+ * interface to create a new embedded misk service:
+ * ```
+ * val embeddedServiceInstance = EmbeddedMisk.create<EmbeddedSampleService>()
+ * ```
+ *
+ * The returned instance is the single access point for the host project to call into the embedded
+ * misk service.
+ */
+object EmbeddedMisk {
+  /**
+   * This is the Application Loader and loads all host project classes and classes depended on by
+   * the embedding interface.
+   */
+  private var hostLoader: ClassLoader = EmbeddedMisk::class.java.classLoader
+
+  inline fun <reified T : Any> create() = create(T::class.java)
+
+  fun <T : Any> create(embeddingInterfaceType: Class<T>): T {
+    val classLoader = createIsolatedLoader(embeddingInterfaceType)
+    val injector = createInjector(classLoader, "${embeddingInterfaceType.name}InjectorFactory")
```

Comment:
Seems like the injectorFactoryClassName construction can be done in createInjector if this just takes `embeddingInterfaceType`?